enclave-runtime/src/tls_ra/tests.rs

```diff
@@ -91,6 +94,13 @@ pub fn test_tls_ra_server_client_networking() {
 
 	assert!(result.is_ok());
 	assert_eq!(*client_shielding_key.read().unwrap(), shielding_key);
-	assert_eq!(*client_state_key.read().unwrap(), state_key);
-	assert_eq!(*client_state.read().unwrap(), state);
+
+	// State and state-key are provisioned only in sidechain mode
+	if WORKER_MODE == WorkerMode::Sidechain {
```

Comment:
We should abstract this worker mode in a trait that we can test independently of the compiler flag.